### PR TITLE
Infra 3287 - Gerrit ECA check plugin issues

### PR DESCRIPTION
**Pull request by autumnfound**
Updated Eclipse Commit listener to make author Gerrit account optional,
removed mentions of LDAP, and added additional checks around the users
identity.

Change-Id: I907fe8a55a07573686d3f79efb6e17f8d56c97e7
Signed-off-by: Martin Lowe <martin.lowe@eclipse-foundation.org>

### CODE REVIEW

**Comment by autumnfound** [comment]
@mbarbero apparently antlr failed, but I'm not sure what that is in the scope of this build.

File: eclipse-cla/src/main/java/org/eclipse/foundation/gerrit/validation/EclipseCommitValidationListener.java [hunk 1]
```diff
@@ -151,37 +146,33 @@ public EclipseCommitValidationListener(PluginConfigFactory cfgFactory) {
...
 		 */
-		IdentifiedUser author = identifyUser(authorIdent);
-		if (author == null) {
-			messages.add(new CommitValidationMessage("The author does not have a Gerrit account.", true));
-			messages.add(new CommitValidationMessage("All authors must either be a commiter on the project, or have a current ECA on file.", false));
-			addDocumentationPointerMessage(messages);
-			addEmptyLine(messages);
-			throw new CommitValidationException("The author must register with Gerrit.", messages);
+		Optional<IdentifiedUser> author = identifyUser(authorIdent);
+		if (author.isPresent()) {
```
Comment (mbarbero) [inline eclipse-cla/src/main/java/org/eclipse/foundation/gerrit/validation/EclipseCommitValidationListener.java]: if NOT author.isPresent

File: eclipse-cla/src/main/java/org/eclipse/foundation/gerrit/validation/EclipseCommitValidationListener.java [hunk 2]
```diff
@@ -211,7 +202,7 @@ public EclipseCommitValidationListener(PluginConfigFactory cfgFactory) {
 		 * Only committers can push on behalf of other users. Note that, I am asking
 		 * if the user (i.e. the person who is doing the actual push) is a committer.
 		 */
-		if (!author.getAccount().equals(user.getAccount())) {
+		if (!author.isPresent() || !author.get().getAccount().equals(user.getAccount())) {
```
Comment (mbarbero) [inline eclipse-cla/src/main/java/org/eclipse/foundation/gerrit/validation/EclipseCommitValidationListener.java]: This is unnecessary as it is already tested above `!author.isPresent() || `

Comment (autumnfound) [reply eclipse-cla/src/main/java/org/eclipse/foundation/gerrit/validation/EclipseCommitValidationListener.java]: Its necessary as the previous check no longer ends the processing, it adds an error message. This way it covers the case if the user doesn't have a user, or if they do and don't match the user.

File: eclipse-cla/src/main/java/org/eclipse/foundation/gerrit/validation/EclipseCommitValidationListener.java (hunk 1 above)
Comment (autumnfound) [reply eclipse-cla/src/main/java/org/eclipse/foundation/gerrit/validation/EclipseCommitValidationListener.java]: Done

**Pull request merged**